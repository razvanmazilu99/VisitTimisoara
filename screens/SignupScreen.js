import React from 'react';
import firebase from '../ApiKeys';
import { View, Text, StyleSheet, TextInput, TouchableOpacity, StatusBar } from 'react-native';

export default class SignupScreen extends React.Component{
    
    constructor(props){
        super(props)

        this.state = ({
            username: '',
            email: '',
            password: ''
        })
    }

    signUpUser = (email, password, username) => {
       try{
            if(this.state.password.length<6){
                alert("Please enter at least 6 characters")
                return;
            }

            firebase.auth().createUserWithEmailAndPassword(email,password).then(user => this.props.navigation.navigate('Login'));

        }catch(error){
            console.log(error.toString());
        }
    }
    
    render(){
        return(
            <View style={styles.container}>
                <StatusBar style="auto" barStyle="light-content" />
                <Text style={styles.registerText}>Create account</Text>
                {/*<TextInput placeholder="Username" onChangeText={(username) => this.setState({username})} style={styles.textInput}></TextInput>*/}
                <TextInput placeholder="Email" onChangeText={(email) => this.setState({email})} style={[styles.textInput, {top: 150}]}></TextInput>
                <TextInput placeholder="Password" onChangeText={(password) => this.setState({password})} secureTextEntry={true} style={[styles.textInput, {top: 170}]}></TextInput>

                <TouchableOpacity onPress={() => this.signUpUser(this.state.email, this.state.password)} style={styles.signupButton}>
                    <Text style={{color: 'white', fontSize: 26, textAlign: 'center', top: 12}}>Sign up</Text>
                </TouchableOpacity>

                <View style={{flex: 1, flexDirection: 'row'}}>
                    <View style={{flex: 0.75}}>
                        <Text style={{ color:'white', fontSize: 18, alignSelf: 'center', left: 18, top: 300}}>Already have an account?</Text>
                    </View>
                
                    <View style={{flex: 0.25}}>
                        <TouchableOpacity onPress={() => this.props.navigation.navigate('Login')} style={styles.textLogIn}>
                            <Text style={{ color:'white', fontSize: 18, alignSelf: 'center', left: -10, color: '#59B298'}}>Log in</Text>
                        </TouchableOpacity>
                    </View>
                </View>

            </View>
        );
    }
}

const styles = StyleSheet.create({

    registerText: {
        top: 80,
        color: 'white',
        fontSize: 40,
    },

    container: {
        flex: 1,
        backgroundColor: '#1A1B29',
        alignItems: 'center',
    },

    textInput: {
        backgroundColor: 'white',
        borderRadius: 20,
        width: 337,
        height: 61,
        top: 130,
        fontSize: 26,
        paddingLeft: 25
    },

    textLogIn: {
        top: 300,
        left: -5,
        fontSize: 18,
        width: 60,
        height: 28
    },

    signupButton: {
        backgroundColor: "#59B398",
        width: 337,
        height: 61,
        borderRadius: 20,
        top: 230
    },

});