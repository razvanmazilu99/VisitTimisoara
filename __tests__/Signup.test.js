import React from 'react';
import SignupScreen from '../screens/SignupScreen';

import { create } from 'react-test-renderer';
import renderer from 'react-test-renderer';

const tree = create(<SignupScreen />);

test('snapshot', () => {
    expect(tree).toMatchSnapshot();
});

it('signup function empty username test', () => {
    let data = renderer.create(<SignupScreen />).getInstance();
    window.alert = jest.fn();
    data.signUpUser('','','');
    
    expect(window.alert).toHaveBeenCalledTimes(1);
    expect(window.alert).toHaveBeenCalledWith('Write your username!');
})

it('signup function empty email test', () => {
    let data = renderer.create(<SignupScreen />).getInstance();
    window.alert = jest.fn();
    data.setState({username: 'username'});
    data.signUpUser('','', data.state.username);
    expect(window.alert).toHaveBeenCalledTimes(1);
    expect(window.alert).toHaveBeenCalledWith('Write your email!');
})

it('signup function empty password test', () => {
    let data = renderer.create(<SignupScreen />).getInstance();
    window.alert = jest.fn();
    data.setState({username: 'username'});
    data.setState({email: 'username@yahoo.com'});
    data.signUpUser(data.state.email, '', data.state.username);
    
    expect(window.alert).toHaveBeenCalledTimes(1);
    expect(window.alert).toHaveBeenCalledWith('Write your password!');
})

it('signup function invalid password test', () => {
    let data = renderer.create(<SignupScreen />).getInstance();
    window.alert = jest.fn();
    data.setState({username: 'username'});
    data.setState({email: 'username@yahoo.com'});
    data.setState({password: '123'});
    data.signUpUser(data.state.email, data.state.password, data.state.username);
    
    expect(window.alert).toHaveBeenCalledTimes(1);
    expect(window.alert).toHaveBeenCalledWith('Password must have at least 6 characters!');
})

it('signup function success test', () => {
    let data = renderer.create(<SignupScreen />).getInstance();
    data.setState({username: 'username'});
    data.setState({email: 'username@yahoo.com'});
    data.setState({password: '123456'});
    data.signUpUser(data.state.email, data.state.password, data.state.username);
    
    expect(data.state.signup).toEqual('success');
})