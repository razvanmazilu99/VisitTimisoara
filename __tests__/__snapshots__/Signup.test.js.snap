// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1A1B29",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 40,
        "top": 80,
      }
    }
  >
    Create account
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Username"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 20,
        "fontSize": 26,
        "height": 61,
        "paddingLeft": 25,
        "top": 130,
        "width": 337,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Email"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderRadius": 20,
          "fontSize": 26,
          "height": 61,
          "paddingLeft": 25,
          "top": 130,
          "width": 337,
        },
        Object {
          "top": 150,
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Password"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderRadius": 20,
          "fontSize": 26,
          "height": 61,
          "paddingLeft": 25,
          "top": 130,
          "width": 337,
        },
        Object {
          "top": 170,
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#59B398",
        "borderRadius": 20,
        "height": 61,
        "opacity": 1,
        "top": 230,
        "width": 337,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 26,
          "textAlign": "center",
          "top": 12,
        }
      }
    >
      Sign up
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 0.75,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontSize": 18,
            "left": 18,
            "top": 300,
          }
        }
      >
        Already have an account?
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 0.25,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "fontSize": 18,
            "height": 28,
            "left": -5,
            "opacity": 1,
            "top": 300,
            "width": 60,
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "#59B298",
              "fontSize": 18,
              "left": -10,
            }
          }
        >
          Log in
        </Text>
      </View>
    </View>
  </View>
</View>
`;
